lib/asset_cloud/asset.rb

```diff
@@ -34,13 +34,8 @@ def self.at(cloud, key, value = nil, metadata = nil, &block)
     end
 
     def <=>(other)
-      return 1 unless other.is_a?(Asset)
-      compare_cloud = cloud.object_id <=> other.cloud.object_id
-      if compare_cloud == 0
-        key <=> other.key
-      else
-        compare_cloud
-      end
+      return 1 unless other.respond_to?(:key)
```

Comment:
From the doc

>  If the other object is not comparable then the <=> operator should return nil.

We should probably return `nil` here

Reply:
You are right @Edouard-chin , I assumed `<=>` always has to return an integer. Will push a fix.